### ISSUE TITLE
Unable to launch Material Editor

### ISSUE THREAD

**Issue by LB-KacperSzczepanski** [o3de/o3de#7113]
**Describe the bug**
Material Editor does not work correctly. Pressing the Material Editor in tools menu results in “(MaterialComponent) - Launching Material Editor” line being printed in the Console, but the tool is not opened. Attempt to launch MaterialEditor.exe application directly leads to a “Negotiation Failed” window being displayed by the Asset Processor. Please refer to the attached video for more information. The issue does not occur in Stabilization/2111RTE installer build.

**Steps to reproduce**
1. Go to Tools > Material Editor.

**Expected behavior**
 The Material Editor tool is opened.
 
**Actual behavior**
 “(MaterialComponent) - Launching Material Editor” line is printed in the console but the tool is not opened and no errors are printed.
 
**Video**

https://user-images.githubusercontent.com/86953282/150805365-542eeb12-fb93-489f-85d8-4051dfd142ce.mp4

**Found in Branch**
Development (878ba24)

**Desktop/Device:**
• Device: PC
• OS: Windows 
• Version: 10 Pro
• CPU: Intel Core i7 8700k
• GPU: Nvidia RTX 2080
• Memory: 16GB

**Comment by gadams3** [comment]
I am unable to reproduce the bug using the latest from the development branch. The commit that was used for testing is around three days old. Could you please reconfirm with the latest? Changes were recently added to prevent the material editor from loading gems that it did not use that affected startup times and performance. If other gems were enabled that depended on any of the disabled gems then that auto load configuration would need to be updated. But that does not seem to be the case here.

Launching the material editor from explorer or the command line requires passing in the active project path or setting the global project in the settings registry.

**Comment by LB-ArturZieba** [comment]
The issue still occurs on the Development (e1cb40f) build (Prebuilt SDK configuration). The project I was reproducing this issue on had all Gems enabled. 

I was unable to launch the Material Editor by going to Tools > Material Editor, using its designated hotkey (M), and by attempting to launch it directly from its .exe via Command Line with the --project-path argument. When attempting to launch the Material Editor via the Command Line with the --project-path argument, there were two Trace::Warnings and one Trace::Error printed. 

Please refer to the attached CMDLog.txt file, and the AttemptingToLaunchMaterialEditor.mp4 video for more information.

[CMDLog.txt](https://github.com/o3de/o3de/files/7932871/CMDLog.txt)

https://user-images.githubusercontent.com/86952082/150962825-91580591-60cc-4133-ba4a-2bc2b4e0a84d.mp4

**Comment by svitlani** [comment]
Hello @gadams3 , can you please update on this ticket status? LB team still reports this issue as active, did you have a chance to verify/fix the issue? Thank you!

**Comment by gadams3** [comment]
Thank you for the additional information. I should not be able to resolve the issue.

Component 'AWSGameLiftClientSystemComponent' is missing another required service: 0x874f91a4
The above message means the application system entity could not initialize because of a missing component service.

The o3de/registry folder has a relatively new file, gems_autoload_materialedtor.setreg. It specifies a list of gems to disable when launching the material editor. Some AWS gems are being disabled that must be required because you have all other gems enabled. I will enable all gems and update the list.

**Comment by gadams3** [comment]
What method are you using to enable all gems?

**Comment by LB-KacperSzczepanski** [comment]
I enable gems by O3DE Project Manager through "Configure Gems..." option in rollup menu next to the project name.

**Comment by gadams3** [comment]
Linking the following feature request which will ensure that gyms that are marked to not auto load will still load if there is some required dependency. #7655 https://github.com/o3de/o3de/issues/7655

**Comment by yuyihsu** [comment]
Is this fixed? remote the RTE and WF5 labels for now so it doesn't impact to our reporting.

**Comment by LB-KacperSzczepanski** [comment]
Issue occurs in Development (146b2c8) when AWS Game Lift gem is enabled.

**Comment by galibzon** [comment]
@yuyihsu , This is an easy fix. We can schedule it for the October relase

**Comment by gadams3** [comment]
@galibzon @yuyihsu @lumberyard-employee-dm This issue occurs because material editor, shader management console, material canvas, have a settings registry file that disables unrelated gems. This was done to improve material editor startup times and resource usage by not initializing gems that won't be used in those tools. However, if the game project requires system components from one of the disabled gems, the system entity won't be able to activate the application won't launch.

The proper fix is blocked by the following feature request https://github.com/o3de/o3de/issues/7655
The interim fix is to update or remove the settings registry files but that will impact startup times and potentially reveal other issues with gems that weren't enabled before.
The issue is currently only being reported with QA enabling every single gem on a project which is bound to expose incompatibilities.

**Comment by yuyihsu** [comment]
This issue happens when all gems are enabled. Currently in the Project Manager we did show some messages reminding user about the gems incompatibility. If user decides to ignore the message, they will have to take this as their own risk. 

Long-term solution will be-
we should give user a much better warning messages and workflow to prevent user from enabling gems that are incompatible with each other.

**Comment by svitlani** [comment]
@gadams3 any update on this? Do you depend on someone else's work being done before you can proceed with fixing this issue? Thanks!

**Comment by LB-KacperKapusta** [comment]
It has been observed in Development (cb09e08) that the Material Editor does not launch for all gems project with GameLift gem disabled. Material Editor launches for project created using Default template. The Default project fails to launch the Material Editor after Motion Matching gem is added, but removing the Motion Matching gem along with GameLift gem from the all gems project does not solve the issue. We were unable to point the exact gem causing the issue due to the time constraints.

**Comment by LB-KacperKapusta** [comment]
Reopening the issue as it has been verified that both the Material Editor and Material Canvas fail to launch whenever the AWS GameLift gem is enabled for a project in Development (d6ef267) and Stabilization/2305 (9e5047f) . However, it has been confirmed that enabling the Motion Matching gem for a project no longer blocks the Material Editor/Material Canvas from launching.

**Comment by gadams3** [comment]
Merged another change to add an exhaustive list of all of the NWS modules and tested locally. Eventually, this will need better messaging for incompatible gems and modules when starting all tools. Currently, the user has to look at the console window or log file and is only told what did not load and a CRC value that needs to be searched for manually.

**Comment by LB-ArturZieba** [comment]
Reopening the issue as it occurs on Development (bc68edc). Material Editor and Material Canvas tools can be launched on a Default Template project, however on a project with all or most Gems enabled it still occurs. Exact Gems causing this were not pinpointed, however this is not caused by AWS GameLift or Motion Matching Gems as before.

**Comment by gadams3** [comment]
Please create a new issue providing material editor or material canvas logs to help identify which modules are having issues. Documentation has also been provided with material editor and material canvas that explains how to remove the registry files that are disabling certain gems as a load time optimization.

**Comment by LB-MichalRodzos** [comment]
The issue can now be tracked in #16795.
